microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/security/keyselection/MultipleConfiguredPublicKeysSelectionTest.java

```diff
@@ -0,0 +1,209 @@
+package org.jboss.eap.qe.microprofile.jwt.security.keyselection;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.CoreMatchers.equalTo;
+
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+
+import javax.json.Json;
+import javax.json.JsonObject;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.OperateOnDeployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.eap.qe.microprofile.jwt.auth.tool.JsonWebToken;
+import org.jboss.eap.qe.microprofile.jwt.auth.tool.JwtHelper;
+import org.jboss.eap.qe.microprofile.jwt.auth.tool.RsaKeyTool;
+import org.jboss.eap.qe.microprofile.jwt.testapp.jaxrs.JaxRsTestApplication;
+import org.jboss.eap.qe.microprofile.jwt.testapp.jaxrs.SecuredJaxRsEndpoint;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * Set of tests verifying key selection functionality. See chapter 9.2.3. JSON Web Key Set (JWKS) of MP-JWT 1.1
```

Comment:
Can you add an information about multiple-deployment scenario?

Reply:
What are you thinking of?

Reply:
Hm, giving it a second thought IMO specifying it in tests' JavaDoc would be better in TP since they are high level customer scenarios. WDYT?

Reply:
I'm sorry, it seems like I am out of the loop. What do you want me to do?

Reply:
After the discussion it has been decided we are good with this